test/common/network/dns_impl_test.cc

```diff
@@ -571,5 +576,23 @@ TEST_P(DnsImplZeroTimeoutTest, Timeout) {
   EXPECT_TRUE(address_list.empty());
 }
 
+// Validate that the resolution timeout timer is enabled if we don't resolve
+// immediately.
+TEST(DnsImplUnitTest, PendingTimerEnable) {
+  Event::MockDispatcher dispatcher;
+  Event::MockTimer* timer = new NiceMock<Event::MockTimer>();
+  EXPECT_CALL(dispatcher, createTimer_(_)).WillOnce(Return(timer));
+  DnsResolverImpl resolver(dispatcher, {});
+  Event::FileEvent* file_event = new NiceMock<Event::MockFileEvent>();
+  EXPECT_CALL(dispatcher, createFileEvent_(_, _, _, _)).WillOnce(Return(file_event));
+  EXPECT_CALL(*timer, disableTimer());
```

Comment:
nit: it would be a better test of the entire test was InSequence or at least the two timer calls.